Compilation failure on Linux - g++

### Operating System

Linux - EndeavourOS x64

### OpenRCT2 build

N/A

### Describe the issue

OpenRCT2 fails to compile on Linux when using g++ due to the following error:

`/home/****/source/openrct2/src/openrct2/drawing/Drawing.String.cpp: In function ‘void ttf_process_string_literal(rct_drawpixelinfo*, std::string_view, text_draw_info*)’:
/home/****/source/openrct2/src/openrct2/drawing/Drawing.String.cpp:839:66: error: ‘*(long unsigned int*)((char*)&ttfRunIndex + offsetof(std::optional<long unsigned int>,std::optional<long unsigned int>::<unnamed>.std::_Optional_base<long unsigned int, true, true>::<unnamed>))’ may be used uninitialized [-Werror=maybe-uninitialized]
  839 |                     auto len = it.GetIndex() - ttfRunIndex.value();
      |                                                                  ^
`

Clang built the program with no problems.

### Area(s) with issue?

Building the game

### Steps to reproduce

1. Install prerequisites
2. make build dir and invoke cmake - make versus ninja is irrelevant
3. Attempt to build OpenRCT2, only to encounter the above error when compiling Drawing.String.cpp


### Attachments

_No response_

> Description of the bug you linked seems different:
> 1. We don't compile in c++20 (yet), the error report claims it is required
> for the bug to trigger 2. The bug report claims there is a compilation
> issue with libstdc++, ours is an unexpected warning 3. I think there is a
> GCC version mismatch, but haven't got the numbers handy

Hm the bugged looked similar from face value.
There are similar bugs from earlier releases, e.g.: https://gcc.gnu.org/
bugzilla/show_bug.cgi?id=92011


Should we report all GCC 12 regressions here?

E.g. I have this:

> In file included from /usr/include/c++/12.1.0/string:40,
> 
>                  from /usr/include/nlohmann/json_fwd.hpp:7,
>                  from
>                  /build/openrct2-git/src/openrct2-git/src/openrct2/manageme
>                  nt/../object/../core/JsonFwd.hpp:12, from
>                  /build/openrct2-git/src/openrct2-git/src/openrct2/manageme
>                  nt/../object/Object.h:13, from
>                  /build/openrct2-git/src/openrct2-git/src/openrct2/manageme
>                  nt/Research.h:13, from
>                  /build/openrct2-git/src/openrct2-git/src/openrct2/manageme
>                  nt/Finance.h:13,> 
>                  from /build/openrct2-git/src/openrct2-git/src/openrct2/
management/Finance.cpp:10:
> In static member function ‘static std::char_traits<char>::char_type*
> std::char_traits<char>::copy(char_type*, const char_type*, std::size_t)’,> 
>     inlined from ‘static void std::__cxx11::basic_string<_CharT, _Traits,
>     _Alloc>::_S_copy(_CharT*, const _CharT*, size_type) [with _CharT =
>     char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
>     at /usr/include/c++/12.1.0/bits/basic_string.h:423:21, inlined from
>     ‘static void std::__cxx11::basic_string<_CharT, _Traits,
>     _Alloc>::_S_copy(_CharT*, const _CharT*, size_type) [with _CharT =
>     char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
>     at /usr/include/c++/12.1.0/bits/basic_string.h:418:7, inlined from
>     ‘std::__cxx11::basic_string<_CharT, _Traits, _Allocator>&
>     std::__cxx11::basic_string<_CharT, _Traits,
>     _Alloc>::_M_replace(size_type, size_type, const _CharT*, size_type)
>     [with _CharT = char; _Traits = std::char_traits<char>; _Alloc =
>     std::allocator<char>]’ at
>     /usr/include/c++/12.1.0/bits/basic_string.tcc:532:22, inlined from
>     ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&
>     std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::assign(const
>     _CharT*) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc
>     = std::allocator<char>]’ at
>     /usr/include/c++/12.1.0/bits/basic_string.h:1647:19, inlined from
>     ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&
>     std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator=(const
>     _CharT*) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc
>     = std::allocator<char>]’ at
>     /usr/include/c++/12.1.0/bits/basic_string.h:815:28,> 
>     inlined from ‘void finance_init()’ at /build/openrct2-git/src/openrct2-
git/src/openrct2/management/Finance.cpp:232:28:
> /usr/include/c++/12.1.0/bits/char_traits.h:431:56: error: ‘void*
> __builtin_memcpy(void*, const void*, long unsigned int)’ accessing
> 9223372036854775810 or more bytes at offsets -4611686018427387902 and
> [-4611686018427387903, 4611686018427387904] may overlap up to
> 9223372036854775813 bytes at offset -3 [-Werror=restrict]> 
>   431 |         return static_cast<char_type*>(__builtin_memcpy(__s1, __s2,
>   __n));>   
>       |                                        ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
>       |                                        ~~~~
> 
> cc1plus: all warnings being treated as errors
> make[2]: *** [CMakeFiles/libopenrct2.dir/build.make:2946:
> CMakeFiles/libopenrct2.dir/src/openrct2/management/Finance.cpp.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> make[1]: *** [CMakeFiles/Makefile2:89: CMakeFiles/libopenrct2.dir/all] Error
> 2 make: *** [Makefile:136: all] Error 2

Another:

> [ 94%] Building CXX object
> CMakeFiles/openrct2.dir/src/openrct2-ui/windows/ObjectLoadError.cpp.o In
> file included from /usr/include/c++/12.1.0/ios:43,
> 
>                  from /usr/include/c++/12.1.0/istream:38,
>                  from /usr/include/c++/12.1.0/fstream:38,
> 
>                  from /build/openrct2-git/src/openrct2-git/src/openrct2-ui/
windows/Changelog.cpp:11:
> In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
> std::basic_streambuf<_CharT, _Traits>::egptr() const [with _CharT = char;
> _Traits = std::char_traits<char>]’,> 
>     inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
>     std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
>     _Traits = std::char_traits<char>]’ at
>     /usr/include/c++/12.1.0/streambuf:324:49, inlined from
>     ‘std::istreambuf_iterator<_CharT, _Traits>&
>     std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
>     char; _Traits = std::char_traits<char>]’ at
>     /usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
>     ‘void std::__cxx11::basic_string<_CharT, _Traits,
>     _Alloc>::_M_construct(_InIterator, _InIterator,
>     std::input_iterator_tag) [with _InIterator =
>     std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
>     _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
>     /usr/include/c++/12.1.0/bits/basic_string.tcc:178:6, inlined from
>     ‘std::__cxx11::basic_string<_CharT, _Traits,
>     _Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
>     [with _InputIterator = std::istreambuf_iterator<char,
>     std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
>     char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
>     at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
>     ‘const std::string ChangelogWindow::GetChangelogText()’ at
>     /build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
>     cpp:74:98,> 
>     inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
> /usr/include/c++/12.1.0/streambuf:493:30: error: potential null pointer
> dereference [-Werror=null-dereference]> 
>   493 |       egptr() const { return _M_in_end; }
>   
>       |                              ^~~~~~~~~
> 
> In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
> std::basic_streambuf<_CharT, _Traits>::gptr() const [with _CharT = char;
> _Traits = std::char_traits<char>]’,> 
>     inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
>     std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
>     _Traits = std::char_traits<char>]’ at
>     /usr/include/c++/12.1.0/streambuf:324:33, inlined from
>     ‘std::istreambuf_iterator<_CharT, _Traits>&
>     std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
>     char; _Traits = std::char_traits<char>]’ at
>     /usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
>     ‘void std::__cxx11::basic_string<_CharT, _Traits,
>     _Alloc>::_M_construct(_InIterator, _InIterator,
>     std::input_iterator_tag) [with _InIterator =
>     std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
>     _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
>     /usr/include/c++/12.1.0/bits/basic_string.tcc:178:6, inlined from
>     ‘std::__cxx11::basic_string<_CharT, _Traits,
>     _Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
>     [with _InputIterator = std::istreambuf_iterator<char,
>     std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
>     char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
>     at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
>     ‘const std::string ChangelogWindow::GetChangelogText()’ at
>     /build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
>     cpp:74:98,> 
>     inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
> /usr/include/c++/12.1.0/streambuf:490:30: error: potential null pointer
> dereference [-Werror=null-dereference]> 
>   490 |       gptr()  const { return _M_in_cur;  }
>   
>       |                              ^~~~~~~~~
> 
> In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
> std::basic_streambuf<_CharT, _Traits>::gptr() const [with _CharT = char;
> _Traits = std::char_traits<char>]’,> 
>     inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
>     std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
>     _Traits = std::char_traits<char>]’ at
>     /usr/include/c++/12.1.0/streambuf:324:33, inlined from
>     ‘std::istreambuf_iterator<_CharT, _Traits>&
>     std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
>     char; _Traits = std::char_traits<char>]’ at
>     /usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
>     ‘void std::__cxx11::basic_string<_CharT, _Traits,
>     _Alloc>::_M_construct(_InIterator, _InIterator,
>     std::input_iterator_tag) [with _InIterator =
>     std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
>     _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
>     /usr/include/c++/12.1.0/bits/basic_string.tcc:205:6, inlined from
>     ‘std::__cxx11::basic_string<_CharT, _Traits,
>     _Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
>     [with _InputIterator = std::istreambuf_iterator<char,
>     std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
>     char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
>     at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
>     ‘const std::string ChangelogWindow::GetChangelogText()’ at
>     /build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
>     cpp:74:98,> 
>     inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
> /usr/include/c++/12.1.0/streambuf:490:30: error: potential null pointer
> dereference [-Werror=null-dereference]> 
>   490 |       gptr()  const { return _M_in_cur;  }
>   
>       |                              ^~~~~~~~~
> 
> In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
> std::basic_streambuf<_CharT, _Traits>::egptr() const [with _CharT = char;
> _Traits = std::char_traits<char>]’,> 
>     inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
>     std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
>     _Traits = std::char_traits<char>]’ at
>     /usr/include/c++/12.1.0/streambuf:324:49, inlined from
>     ‘std::istreambuf_iterator<_CharT, _Traits>&
>     std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
>     char; _Traits = std::char_traits<char>]’ at
>     /usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
>     ‘void std::__cxx11::basic_string<_CharT, _Traits,
>     _Alloc>::_M_construct(_InIterator, _InIterator,
>     std::input_iterator_tag) [with _InIterator =
>     std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
>     _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
>     /usr/include/c++/12.1.0/bits/basic_string.tcc:205:6, inlined from
>     ‘std::__cxx11::basic_string<_CharT, _Traits,
>     _Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
>     [with _InputIterator = std::istreambuf_iterator<char,
>     std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
>     char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
>     at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
>     ‘const std::string ChangelogWindow::GetChangelogText()’ at
>     /build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
>     cpp:74:98,> 
>     inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
> /usr/include/c++/12.1.0/streambuf:493:30: error: potential null pointer
> dereference [-Werror=null-dereference]> 
>   493 |       egptr() const { return _M_in_end; }
>   
>       |                              ^~~~~~~~~
> 
> cc1plus: all warnings being treated as errors
> make[2]: *** [CMakeFiles/openrct2.dir/build.make:804:
> CMakeFiles/openrct2.dir/src/openrct2-ui/windows/Changelog.cpp.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> make[1]: *** [CMakeFiles/Makefile2:167: CMakeFiles/openrct2.dir/all] Error 2
> make: *** [Makefile:136: all] Error 2

```
[ 94%] Building CXX object
CMakeFiles/openrct2.dir/src/openrct2-ui/windows/ObjectLoadError.cpp.o In
file included from /usr/include/c++/12.1.0/ios:43,
 
from /usr/include/c++/12.1.0/istream:38,
from /usr/include/c++/12.1.0/fstream:38,
 
from /build/openrct2-git/src/openrct2-git/src/openrct2-ui/
windows/Changelog.cpp:11:
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::egptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:49, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:178:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:493:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
493 |       egptr() const { return _M_in_end; }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::gptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:33, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:178:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:490:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
490 |       gptr()  const { return _M_in_cur;  }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::gptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:33, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:205:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:490:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
490 |       gptr()  const { return _M_in_cur;  }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::egptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:49, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:205:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:493:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
493 |       egptr() const { return _M_in_end; }
   
       |                              ^~~~~~~~~
 
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/openrct2.dir/build.make:804:
CMakeFiles/openrct2.dir/src/openrct2-ui/windows/Changelog.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:167: CMakeFiles/openrct2.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

`[ 94%] Building CXX object
CMakeFiles/openrct2.dir/src/openrct2-ui/windows/ObjectLoadError.cpp.o In
file included from /usr/include/c++/12.1.0/ios:43,
 
from /usr/include/c++/12.1.0/istream:38,
from /usr/include/c++/12.1.0/fstream:38,
 
from /build/openrct2-git/src/openrct2-git/src/openrct2-ui/
windows/Changelog.cpp:11:
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::egptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:49, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:178:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:493:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
493 |       egptr() const { return _M_in_end; }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::gptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:33, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:178:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:490:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
490 |       gptr()  const { return _M_in_cur;  }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::gptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:33, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:205:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:490:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
490 |       gptr()  const { return _M_in_cur;  }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::egptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:49, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:205:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:493:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
493 |       egptr() const { return _M_in_end; }
   
       |                              ^~~~~~~~~
 
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/openrct2.dir/build.make:804:
CMakeFiles/openrct2.dir/src/openrct2-ui/windows/Changelog.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:167: CMakeFiles/openrct2.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
`

```
[ 94%] Building CXX object
CMakeFiles/openrct2.dir/src/openrct2-ui/windows/ObjectLoadError.cpp.o In
file included from /usr/include/c++/12.1.0/ios:43,
 
from /usr/include/c++/12.1.0/istream:38,
from /usr/include/c++/12.1.0/fstream:38,
 
from /build/openrct2-git/src/openrct2-git/src/openrct2-ui/
windows/Changelog.cpp:11:
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::egptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:49, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:178:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:493:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
493 |       egptr() const { return _M_in_end; }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::gptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:33, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:178:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:490:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
490 |       gptr()  const { return _M_in_cur;  }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::gptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:33, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:205:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:490:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
490 |       gptr()  const { return _M_in_cur;  }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::egptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:49, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:205:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:493:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
493 |       egptr() const { return _M_in_end; }
   
       |                              ^~~~~~~~~
 
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/openrct2.dir/build.make:804:
CMakeFiles/openrct2.dir/src/openrct2-ui/windows/Changelog.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:167: CMakeFiles/openrct2.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

I guess Github just butchers any email they get.

2nd issue:
```

In file included from /usr/include/c++/12.1.0/string:40,
 
from /usr/include/nlohmann/json_fwd.hpp:7,
from
/build/openrct2-git/src/openrct2-git/src/openrct2/manageme
nt/../object/../core/JsonFwd.hpp:12, from
/build/openrct2-git/src/openrct2-git/src/openrct2/manageme
nt/../object/Object.h:13, from
/build/openrct2-git/src/openrct2-git/src/openrct2/manageme
nt/Research.h:13, from
/build/openrct2-git/src/openrct2-git/src/openrct2/manageme
nt/Finance.h:13,> 
from /build/openrct2-git/src/openrct2-git/src/openrct2/
management/Finance.cpp:10:
In static member function ‘static std::char_traits<char>::char_type*
std::char_traits<char>::copy(char_type*, const char_type*, std::size_t)’,> 
inlined from ‘static void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_S_copy(_CharT*, const _CharT*, size_type) [with _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:423:21, inlined from
‘static void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_S_copy(_CharT*, const _CharT*, size_type) [with _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:418:7, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits, _Allocator>&
std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_replace(size_type, size_type, const _CharT*, size_type)
[with _CharT = char; _Traits = std::char_traits<char>; _Alloc =
std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:532:22, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&
std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::assign(const
_CharT*) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc
= std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.h:1647:19, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&
std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator=(const
_CharT*) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc
= std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.h:815:28,> 
inlined from ‘void finance_init()’ at /build/openrct2-git/src/openrct2-
git/src/openrct2/management/Finance.cpp:232:28:
/usr/include/c++/12.1.0/bits/char_traits.h:431:56: error: ‘void*
__builtin_memcpy(void*, const void*, long unsigned int)’ accessing
9223372036854775810 or more bytes at offsets -4611686018427387902 and
[-4611686018427387903, 4611686018427387904] may overlap up to
9223372036854775813 bytes at offset -3 [-Werror=restrict]> 
431 | return static_cast<char_type*>(__builtin_memcpy(__s1, __s2,
__n));> 
 | ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
 | ~~~~
 
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/libopenrct2.dir/build.make:2946:
CMakeFiles/libopenrct2.dir/src/openrct2/management/Finance.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:89: CMakeFiles/libopenrct2.dir/all] Error
2 make: *** [Makefile:136: all] Error 2
```

Another:


```
[ 94%] Building CXX object
CMakeFiles/openrct2.dir/src/openrct2-ui/windows/ObjectLoadError.cpp.o In
file included from /usr/include/c++/12.1.0/ios:43,
 
from /usr/include/c++/12.1.0/istream:38,
from /usr/include/c++/12.1.0/fstream:38,
 
from /build/openrct2-git/src/openrct2-git/src/openrct2-ui/
windows/Changelog.cpp:11:
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::egptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:49, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:178:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:493:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
493 |       egptr() const { return _M_in_end; }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::gptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:33, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:178:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:490:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
490 |       gptr()  const { return _M_in_cur;  }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::gptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:33, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:205:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:490:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
490 |       gptr()  const { return _M_in_cur;  }
   
       |                              ^~~~~~~~~
 
In member function ‘std::basic_streambuf<_CharT, _Traits>::char_type*
std::basic_streambuf<_CharT, _Traits>::egptr() const [with _CharT = char;
_Traits = std::char_traits<char>]’,> 
inlined from ‘std::basic_streambuf<_CharT, _Traits>::int_type
std::basic_streambuf<_CharT, _Traits>::sbumpc() [with _CharT = char;
_Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/streambuf:324:49, inlined from
‘std::istreambuf_iterator<_CharT, _Traits>&
std::istreambuf_iterator<_CharT, _Traits>::operator++() [with _CharT =
char; _Traits = std::char_traits<char>]’ at
/usr/include/c++/12.1.0/bits/streambuf_iterator.h:172:17, inlined from
‘void std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::_M_construct(_InIterator, _InIterator,
std::input_iterator_tag) [with _InIterator =
std::istreambuf_iterator<char, std::char_traits<char> >; _CharT = char;
_Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at
/usr/include/c++/12.1.0/bits/basic_string.tcc:205:6, inlined from
‘std::__cxx11::basic_string<_CharT, _Traits,
_Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&)
[with _InputIterator = std::istreambuf_iterator<char,
std::char_traits<char> >; <template-parameter-2-2> = void; _CharT =
char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
at /usr/include/c++/12.1.0/bits/basic_string.h:756:16, inlined from
‘const std::string ChangelogWindow::GetChangelogText()’ at
/build/openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.
cpp:74:98,> 
inlined from ‘bool ChangelogWindow::ReadChangelogFile()’ at /build/
openrct2-git/src/openrct2-git/src/openrct2-ui/windows/Changelog.cpp:265:46:
/usr/include/c++/12.1.0/streambuf:493:30: error: potential null pointer
dereference [-Werror=null-dereference]> 
493 |       egptr() const { return _M_in_end; }
   
       |                              ^~~~~~~~~
 
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/openrct2.dir/build.make:804:
CMakeFiles/openrct2.dir/src/openrct2-ui/windows/Changelog.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:167: CMakeFiles/openrct2.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

--- a/src/openrct2/drawing/Drawing.String.cpp
+++ b/src/openrct2/drawing/Drawing.String.cpp
@@ -843,14 +843,14 @@ static void TTFProcessStringLiteral(DrawPixelInfo& dpi, std::string_view text, T
                     // Draw the TTF run
                     // This error suppression abomination is here to suppress https://github.com/OpenRCT2/OpenRCT2/issues/17371.
                     // Additionally, we have to suppress the error for the error suppression... :'-(
-                    // TODO: Re-evaluate somewhere in 2023.
-#    ifdef __MINGW32__
+                    // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105937 is fixed in GCC13
+#    if defined(__GNUC__) && !defined(__clang__)
 #        pragma GCC diagnostic push
 #        pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
 #    endif
                     auto len = it.GetIndex() - ttfRunIndex.value();
                     TTFDrawStringRawTTF(dpi, text.substr(ttfRunIndex.value(), len), info);
-#    ifdef __MINGW32__
+#    if defined(__GNUC__) && !defined(__clang__)
 #        pragma GCC diagnostic pop
 #    endif
                     ttfRunIndex = std::nullopt;

